### PR TITLE
feat(wmg): wmg v3 api

### PR DESCRIPTION
**Pull request by atarashansky**
## Reason for Change

- #6065
- #6071  

## Changes

- added dynamic scaling to legend and colorscale
- updated backend to read v3 snapshot
- Re-activated skipped unit tests now that v3 has been activated

## Dynamic scaling demo 
https://czi-sci.slack.com/archives/C0247APK621/p1698163069890259
## Testing steps

- All tests pass, including previously skipped tests
- E2E tests pass

### CODE REVIEW

File: frontend/src/views/WheresMyGeneV2/components/Main/index.tsx [hunk 1]
```diff
@@ -158,7 +167,42 @@ export default function WheresMyGene(): JSX.Element {
           const { cellTypeGeneExpressionSummaries } = geneExpressionSummary;
 
           for (const cellTypeGeneExpressionSummary of cellTypeGeneExpressionSummaries) {
+            // get term before $
+            const cellTypeGeneExpressionSummaryId =
+              cellTypeGeneExpressionSummary.viewId.split("$")[0];
```
Comment (tihuan) [inline frontend/src/views/WheresMyGeneV2/components/Main/index.tsx:172]: you can use `getOntologyTermIdFromCellTypeViewId(viewId)`!

File: frontend/src/views/WheresMyGeneV2/components/Main/index.tsx [hunk 2]
```diff
@@ -158,7 +167,42 @@ export default function WheresMyGene(): JSX.Element {
...
+
+            /*
+            This is to dynamically set the min and max based on the data that is visible to users.
+            Skip conditions:
+            - If the cell type is not part of an expanded tissue
+            - If the tissue does not contain any of the filtered cell types and at least one cell type is filtered
+            - If the cell type is not included in the filtered cell types and at least one cell type is filtered
+            - If the tissue id is not included in the filtered tissue ids and at least one tissue is filtered
+            - If the cell type gene expression summary view id is not included in the available view ids
+            The above conditions capture all possible scenarios where the data is not visible to users.
```
Comment (tihuan) [inline frontend/src/views/WheresMyGeneV2/components/Main/index.tsx:184]: awesome documentation 🙏 Thanks so much!

File: frontend/src/views/WheresMyGene/components/HeatMap/index.tsx [hunk 3]
```diff
@@ -255,6 +255,7 @@ export default memo(function HeatMap({
                 echartsRendererMode={echartsRendererMode}
                 setAllChartProps={setAllChartProps}
                 chartProps={allChartProps[tissue]}
+                maxExpression={6.0}
```
Comment (tihuan) [inline frontend/src/views/WheresMyGene/components/HeatMap/index.tsx:258]: can we add some comment in code why we use 6.0 here even though we're not using RankIt anymore? Thanks!

**Review by tihuan — COMMENTED** [review]
Looks really great! Thanks for fixing the scaling, @atarashansky 🙏 

Can you add the videos you have in Slack in this PR for paper trail please? Since Slack messages don't stay around for long

Also any e2e tests you can think of we should add?

Thank you!

File: frontend/src/views/WheresMyGene/components/HeatMap/index.tsx (hunk 3 above)
Comment (atarashansky) [reply frontend/src/views/WheresMyGene/components/HeatMap/index.tsx:258]: This component is no longer used, so I didn't think any comments were necessary - when are we deleting it?

**Comment by atarashansky** [comment]
> Also any e2e tests you can think of we should add?

This seems like it would be better covered by FE unit tests. Do we have any test framework set up to run those?

File: backend/wmg/data/snapshot.py [hunk 4]
```diff
@@ -73,8 +72,12 @@ class WmgSnapshot:
     filter_relationships: Optional[Dict] = field(default=None)
 
     # dataset metadata dictionary
+
     dataset_metadata: Optional[Dict] = field(default=None)
 
+    def __hash__(self):
```
Comment (prathapsridharan) [inline backend/wmg/data/snapshot.py:78]: if the object doesn't have to hashable then why is this there? Could it be removed?

**Review by kaloster — APPROVED** [review]

**Review by prathapsridharan — APPROVED** [review]
Looks good. Just a small comment about removing `__hash__` on `WMGSnapshot` if it is not needed

File: frontend/src/views/WheresMyGene/components/HeatMap/index.tsx (hunk 3 above)
Comment (tihuan) [reply frontend/src/views/WheresMyGene/components/HeatMap/index.tsx:258]: ah I see, so 6.0 is just for passing TS. @kaloster is working on removing WMGv1!

**Comment by tihuan** [comment]
> > Also any e2e tests you can think of we should add?
> 
> This seems like it would be better covered by FE unit tests. Do we have any test framework set up to run those? In any case, we can write an E2E test that just makes sure that the unscaled max value is changing with the filters.

CC has unit test examples here 👍 

https://github.com/chanzuckerberg/single-cell-data-portal/blob/cb44c8c395ff73df23060ec43f5a6b488e6c82d5/frontend/tests/features/collection/collection.test.ts#L35

**Pull request merged**